test/multioption.test.ts

```diff
@@ -0,0 +1,122 @@
+import { expect, test } from 'vitest';
+import { array, string } from '../src/types';
+import { multioption } from '../src/multioption';
+import { tokenize } from '../src/newparser/tokenizer';
+import { createRegisterOptions } from './createRegisterOptions';
+import { parse } from '../src/newparser/parser';
+import * as Result from '../src/Result';
+
+test('applies default value when no option is provided', async () => {
+  const argv = '';
+  const tokens = tokenize(argv.split(' '));
+  const argparser = multioption({
+    type: array(string),
+    long: 'hello',
+    defaultValue: () => ['world!'],
+    description: 'description',
+  });
+  const registerOptions = createRegisterOptions();
+  argparser.register(registerOptions);
+  const nodes = parse(tokens, registerOptions);
+
+  const result = argparser.parse({
+    nodes,
+    visitedNodes: new Set(),
+  });
+
+  await expect(result).resolves.toEqual(Result.ok(['world!']));
+});
+
+test('does not apply default value when option is provided', async () => {
+  const argv = '--hello=moshe';
+  const tokens = tokenize(argv.split(' '));
+  const argparser = multioption({
+    type: array(string),
+    long: 'hello',
+    defaultValue: () => ['world!'],
+    description: 'description',
+  });
+  const registerOptions = createRegisterOptions();
+  argparser.register(registerOptions);
+  const nodes = parse(tokens, registerOptions);
+
+  const result = argparser.parse({
+    nodes,
+    visitedNodes: new Set(),
+  });
+
+  await expect(result).resolves.toEqual(Result.ok(['moshe']));
+});
+
+test('does not apply default value when options are provided', async () => {
+  const argv = '--hello=moshe --hello=haim';
+  const tokens = tokenize(argv.split(' '));
+  const argparser = multioption({
+    type: array(string),
+    long: 'hello',
+    defaultValue: () => ['world!'],
+    description: 'description',
+  });
+  const registerOptions = createRegisterOptions();
+  argparser.register(registerOptions);
+  const nodes = parse(tokens, registerOptions);
+
+  const result = argparser.parse({
+    nodes,
+    visitedNodes: new Set(),
+  });
+
+  await expect(result).resolves.toEqual(Result.ok(['moshe', 'haim']));
+});
+
+test('fails when no option is provided and applying default value fails', async () => {
+  const argv = '';
+  const tokens = tokenize(argv.split(' '));
+  const argparser = multioption({
+    type: array(string),
+    long: 'hello',
+    defaultValue: () => {
+      throw new Error('its too hot outside, stay inside muthafucka!');
```

Comment:
Please don't use swear words in the codebase <picture data-single-emoji=":blob-angel:" title=":blob-angel:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/blob-angel/176c0e2df4ebf945.gif" alt=":blob-angel:" align="absmiddle"></picture>

Reply:
ok.. 😞